JsonPatch.Tests/ExtensionTests.cs

```diff
@@ -0,0 +1,212 @@
+			e1 = op.ApplyPatch(e1);
+			Assert.AreEqual(e1.Id, e2.Id);
+		}
+		
+		
+		[Test]
+		public void AddArray_Test()
+		{
+			var e1 = JsonDocument.Parse("[1,2,3]");
+			var e2 = JsonDocument.Parse("[1,2,3,4]");
+			var op = e1.CreatePatch(e2, new JsonSerializerOptions() {IgnoreNullValues = true});
+			Assert.AreEqual(op.Operations.Count, 1);
+			Assert.AreEqual(op.Operations[0].Op, OperationType.Add);
+
+			Assert.AreNotEqual(e1.RootElement.ToString(),e2.RootElement.ToString());
+			e1 = op.ApplyPatch(e1);
+			Assert.AreEqual(e1.RootElement.ToString(),e2.RootElement.ToString());
+		}
+
+		[Test]
+		public void RemoveArray_Test()
+		{
+			var e1 = JsonDocument.Parse("[1,2,3]");
+			var e2 = JsonDocument.Parse("[1,2]");
+			var op = e1.CreatePatch(e2, new JsonSerializerOptions() {IgnoreNullValues = true});
+			Assert.AreEqual(op.Operations.Count, 1);
+			Assert.AreEqual(op.Operations[0].Op, OperationType.Remove);
+
+			Assert.AreNotEqual(e1.RootElement.ToString(),e2.RootElement.ToString());
+			e1 = op.ApplyPatch(e1);
+			Assert.AreEqual(e1.RootElement.ToString(),e2.RootElement.ToString());
+		}
+
+		[Test]
+		public void ReplaceArray_Test()
+		{
+			var e1 = JsonDocument.Parse("[1,2,3]");
+			var e2 = JsonDocument.Parse("[1,2,1]");
+			var op = e1.CreatePatch(e2, new JsonSerializerOptions() {IgnoreNullValues = true});
+			Assert.AreEqual(op.Operations.Count, 1);
+			Assert.AreEqual(op.Operations[0].Op, OperationType.Replace);
+
+			Assert.AreNotEqual(e1.RootElement.ToString(),e2.RootElement.ToString());
+			e1 = op.ApplyPatch(e1);
+			Assert.AreEqual(e1.RootElement.ToString(),e2.RootElement.ToString());
+		}
+
+
+		[Test]
+		public void ComplexObject_Test()
+		{
+			var e1 = new TestModel()
+			{
+				Id = Guid.NewGuid(),
+				Attributes = JsonDocument.Parse("[{\"test\":\"test123\"},{\"test\":\"test32132\"},{\"test1\":\"test321\"},{\"test\":[1,2,3]},{\"test\":[1,2,3]}]"),
+				Name = "Test",
+				Numbers = new[] {1, 2, 3},
+				Strings = new[] {"test1", "test2"},
+				InnerObjects = new List<TestModel>() {new TestModel() {Id = Guid.NewGuid(), Name = "TestNameInner1", Numbers = new[] {3, 2, 1}, Strings = new[] {"Test3", "test4"}}}
+			};
+			var e1Json = JsonSerializer.Serialize(e1);
+			var e2 = new TestModel()
+			{
+				Id = Guid.NewGuid(),
+				Attributes = JsonDocument.Parse("[{\"test1\":\"test123\"},{\"test\":\"test32132\"},{\"test1\":\"test321\"},{\"test\":[1,1,3]}]"),
+				Name = "Test4",
+				Numbers = new[] {1, 2, 3, 4},
+				Strings = new[] {"test2", "test2"},
+				InnerObjects = new List<TestModel>() {new TestModel() {Id = Guid.NewGuid(), Name = "TestNameInner1", Numbers = new[] {1, 2, 1}, Strings = new[] {"Test3", "test4", "test5"}}}
+			};
+			var e2Json = JsonSerializer.Serialize(e2);
+			var op = e1.CreatePatch(e2, new JsonSerializerOptions() {IgnoreNullValues = true});
+			var opBack = e2.CreatePatch(e1, new JsonSerializerOptions() {IgnoreNullValues = true});
+
+			Assert.AreNotEqual(e1, e2);
+			e1 = op.ApplyPatch(e1);
```

Comment:
Let's not reuse `e1`.  Define a new variable, please.  This lets you perform a comparison between the two objects directly instead of what's currently on line 176.

This goes for the other tests as well.